pkg/ingester/user_metrics_metadata.go

```diff
@@ -98,18 +98,31 @@ func (mm *userMetricsMetadata) toClientMetadata(req *client.MetricsMetadataReque
 		rCap = req.Limit
 	}
 	r := make([]*mimirpb.MetricMetadata, 0, rCap)
+	if req.Metric != "" {
+		if req.Limit == 0 {
+			return r
+		}
+		set := mm.metricToMetadata[req.Metric]
+		var lengthPerMetric int32
+		for m := range set {
+			if req.LimitPerMetric > 0 && lengthPerMetric >= req.LimitPerMetric {
+				break
+			}
+			m := m
+			r = append(r, &m)
+			lengthPerMetric++
+		}
```

Comment:
I would move this logic to an utility function, so that we can use it below too.